### PR TITLE
[Data Model] Rename EMBER_ZCL to MATTER_CL

### PR DESCRIPTION
**Pull request by jepenven-silabs**
Ember prefixes are legacy from the ember corporation that was acquired a while back by Silicon Labs. 

This is PR 1 of 3 to replace the `EMBER` prefix in the Matter stack defines with matter related defines.

In this PR `EMBER_ZCL` which stands for `Ember_ZigbeeClusterLibrary` are replaces with `MATTER_CL` which stands for `Matter_ClusterLibrary`

This was done by using VSCode Search in replace. 
Term searched (with case sensitivity): `EMBER_ZCL_` 
replaced with `MATTER_CL_`
exclusion path : `./.environment/*, ./out/*, ./third_party/*`

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]
I trust the CI 👀 . 
I did pay attention to the .zapt changes, Accessors.cpp and what is under util.

I am also fine with the `MATTER_CL` so it should  all be fine.

**Comment by andy31415** [comment]
This is potential public API - It may be disruptive to people for 1.3 integration in case these ZCL statuses are still used (especially for accessors ... for most other things CHIP_ERROR should be used)

File: src/app/app-platform/ContentApp.cpp [hunk 1]
```diff
@@ -48,7 +48,7 @@ EmberAfStatus ContentApp::HandleReadAttribute(ClusterId clusterId, AttributeId a
                     "Read Attribute for endpoint " ChipLogFormatMEI " cluster " ChipLogFormatMEI " attribute " ChipLogFormatMEI,
                     ChipLogValueMEI(mEndpointId), ChipLogValueMEI(clusterId), ChipLogValueMEI(attributeId));
 
-    return EMBER_ZCL_STATUS_FAILURE;
+    return MATTER_CL_STATUS_FAILURE;
```
Comment (bzbarsky-apple) [inline src/app/app-platform/ContentApp.cpp:51]: Realistically... we should be replacing these with InteractionModel::Status.  Renaming is fine, but the name is not the major problem here.

Comment (jepenven-silabs) [reply src/app/app-platform/ContentApp.cpp:51]: agreed. However replacing an error code with another one is a less trivial undertaking. Can we settle for a renaming in this PR and move the replacement of `EmberAFStatus` to `InteractionModel::Status` in a subsequent PR ?

Comment (bzbarsky-apple) [reply src/app/app-platform/ContentApp.cpp:51]: Sure.  Just not sure what the win is from this PR then, and how we justify the churn and potential API compat breaks....

**Comment by jepenven-silabs** [comment]
Renaming the define isn't the right solution. Will replace EmberAfStatus altogether with the InteractionModel::Status

**Pull request closed without merging**